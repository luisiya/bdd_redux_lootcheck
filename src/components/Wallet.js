import React, { Component } from "react";
import { connect } from "react-redux";
import { deposit, withdraw } from "../actions/balance";

export class Wallet extends Component{
    constructor(props) {
        super(props);
        this.state = { balance: undefined};
    }
    updateBalance = event => this.setState({balance: parseInt(event.target.value)});
    openDeposit = () => this.props.deposit(this.state.balance);
    withdrawFromDeposit = () => this.props.withdraw(this.state.balance);

    render(){
        return(
            <div>
                <h3 className="balance">Wallet Balance : {this.props.balance}</h3>
                <br/>
                <input className="input-wallet" onChange={this.updateBalance}/>
                <button className="btn-deposit" onClick={this.openDeposit}>Add to Deposit</button>
                <button className="btn-withdraw" onClick={this.withdrawFromDeposit}>Withdraw</button>
            </div>
        )
    }
}

export default connect(state => { return {balance: state.balance} }, {deposit, withdraw})(Wallet);