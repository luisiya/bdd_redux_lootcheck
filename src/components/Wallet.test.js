import React from "react";
import { shallow } from "enzyme";
import { Wallet } from "./Wallet";

describe("Wallet", () => {
    const mockDeposit = jest.fn();
    const mockWithdraw = jest.fn();
    const props = { balance: 20, deposit: mockDeposit, withdraw: mockWithdraw };
    const wallet = shallow(<Wallet {...props} />);

    it("Wallet renders correctly", () => {
        expect(wallet).toMatchSnapshot();
    });
    it("Correctly display balance from props", () => {
        expect(wallet.find(".balance").text()).toEqual("Wallet Balance : 20")
    });
    it("Check of input for balance existing", () => {
        expect(wallet.find(".input-wallet").exists()).toEqual(true);
    });
    describe("When user type some text to input", () => {
        const userBalance = 25;

        beforeEach(() => {
            wallet.find(".input-wallet").simulate("change", {target: {value: userBalance}});
        });

        it("Update balance in state and convert it", () => {
            expect(wallet.state().balance).toEqual(parseInt(userBalance, 10));
        });
        describe("User wants to make deposit", () => {
            beforeEach(() => wallet.find(".btn-deposit").simulate("click"));

            it("dispatches deposit from props", () => {
                expect(mockDeposit).toHaveBeenCalledWith(parseInt(userBalance, 10));
            });
        });
        describe('User wants to withdraw from deposit', function () {
            beforeEach(() => wallet.find(".btn-withdraw").simulate("click"));

            it("dispatches withdraw from props", () => {
                expect(mockWithdraw).toHaveBeenCalledWith(parseInt(userBalance, 10));
            });
        });

    });

});