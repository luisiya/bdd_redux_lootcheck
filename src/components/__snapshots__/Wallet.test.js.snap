// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet Wallet renders correctly 1`] = `ShallowWrapper {}`;
